File: biblioteca/src/interfaces/Reserva.ts
import { Livro } from "./Livro";
import { Usuario } from "./Usuario";

export interface Reserva {
    reservaId: number;
    livroId: string;
    livro?: Livro;
    usuarioId: number;
    usuario?: Usuario;
    dataReserva: Date;
    dataExpiracao: Date;
}

File: Projeto/Models/Reserva.cs
using System;

namespace Projeto.Models;
    public class Reserva
    {
        public int ReservaId { get; set; }
        public String? LivroId { get; set; }
        public Livro? Livro { get; set; }
        public int UsuarioId { get; set; }
        public Usuario? Usuario { get; set; }
        public DateTime DataReserva { get; set; }
        public DateTime? DataExpiracao { get; set; }

        public Reserva()
        {
            DataReserva = DateTime.Now;
            DataExpiracao = DataReserva.AddDays(7);
        }
    }